server/services/applicationService.ts

```diff
@@ -81,13 +82,17 @@ export default class ApplicationService {
     if (Object.keys(request.body).length) {
       return request.body
     }
-    return this.getSessionData(request)
+    return this.getSessionDataForPage(request)
   }
 
-  private getSessionData(request: Request) {
+  private getAllSessionData(request: Request) {
     const data = this.fetchOrInitializeSessionData(request.session, request.params.task, request.params.id)
 
-    return data.application[request.params.id][request.params.task][request.params.page] || {}
+    return data.application[request.params.id] || {}
+  }
+
+  private getSessionDataForPage(request: Request) {
+    return this.getAllSessionData(request)[request.params.task][request.params.page] || {}
```

Comment:
Probably not for now, but just flagging that we did discuss flattening the structure of the session data. I think it might still be worth keeping the task, but maybe not the page.

Reply:
Yeah I did wonder if it would be best to consolidate the two objects that we pass into the Page constructor as the session is just a larger version of the body but I think it's best we wait until we have the flattened structure.